### ISSUE TITLE
Localkube fails to start - no functional network inside minikube VM (KVM)

### ISSUE THREAD

**Issue by firecat53** [kubernetes/minikube#1841]
BUG REPORT

Please provide the following details:

**Environment**:

**Minikube version** (use `minikube version`): minikube version: v0.21.0                                                                                                              
- **OS** (e.g. from /etc/os-release): Archlinux
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): kvm
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): minikube-v0.23.0.iso
- **Install tools**: Installed docker-machine-kvm (tried both version 0.7.0 and 0.10.0), kubectl-bin(1.7.3)  and minikube from Arch AUR
- **Others**:

**What happened**:

Summary: localkube fails to start because networking inside the minikube virtual machine isn't working properly and the apiserver fails to start.
    
    minikube --vm-driver=kvm -v9

    apiserver: Exit with error: Unable to find suitable network address.error='Unable to select an IP.'. Try to set the AdvertiseAddress directly or provide a valid BindAddress to fix this. 

I'm able to ssh into the VM but once inside, there is no available network.

    ping 8.8.8.8
    PING 8.8.8.8 (8.8.8.8): 56 data bytes                                                                                                  
    ping: sendto: Network is unreachable
    ping google.com
    ping: bad address 'google.com'                                                                                                         

**What you expected to happen**:

I'd expect localkube to start and run without errors and that networking was functional from inside the minikub VM.

**How to reproduce it** (as minimally and precisely as possible):

- Install and configure libvirt, docker and docker-machine from Arch repos. These were all functioning normally on their own (I have docker containers running and other kvm virtual machines have no networking issues).
- Install docker-machine-kvm driver from a modified PKGBUILD to get the latest 0.10.0 version. Initially used the 0.7.0 version from the AUR, but neither version changed the outcome.
- Run the `kvm start` command above
- `minikube status` will show localkube as stopped

**Output of `minikube logs` (if applicable)**:

[journalctl - gist](https://gist.github.com/f45785f7bc9e580569ce35ec97e8ebd1)
[Debug logs - gist](https://gist.github.com/firecat53/6f4960b8d3f3503b12082dbde95d58b4)

    $ journalctl -u systemd-networkd                                                                                                       
    -- Logs begin at Fri 2017-08-18 16:41:03 UTC, end at Fri 2017-08-18 17:07:36 UTC. --                                                   
    Aug 18 16:41:03 minikube systemd[1]: Starting Network Service...
    Aug 18 16:41:03 minikube systemd-networkd[1974]: Enumeration completed
    Aug 18 16:41:03 minikube systemd[1]: Started Network Service.
    Aug 18 16:41:03 minikube systemd-networkd[1974]: eth1: IPv6 enabled for interface: Success
    Aug 18 16:41:03 minikube systemd-networkd[1974]: eth0: IPv6 enabled for interface: Success
    Aug 18 16:41:04 minikube systemd-networkd[1974]: eth1: Gained carrier
    Aug 18 16:41:04 minikube systemd-networkd[1974]: eth0: Gained carrier
    Aug 18 16:41:05 minikube systemd-networkd[1974]: eth0: Gained IPv6LL
    Aug 18 16:41:05 minikube systemd-networkd[1974]: eth1: Gained IPv6LL
    Aug 18 16:41:07 minikube systemd-networkd[1974]: eth1: DHCPv4 address 192.168.42.247/24
    $ journalctl -u systemd-resolved                                                                                                       
    -- Logs begin at Fri 2017-08-18 16:41:03 UTC, end at Fri 2017-08-18 17:08:47 UTC. --                                                   
    Aug 18 16:41:03 minikube systemd[1]: Starting Network Name Resolution...
    Aug 18 16:41:04 minikube systemd-resolved[2057]: Positive Trust Anchors:
    Aug 18 16:41:04 minikube systemd-resolved[2057]: . IN DS 19036 8 2 49aac11d7b6f6446702e54a1607371607a1a41855200fd2ce1cdde32f24e8fb5
    Aug 18 16:41:04 minikube systemd-resolved[2057]: Negative trust anchors: 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.arpa 19.172.in-addr.arpa 20.172.in-addr.arpa 21.172.in-addr.arpa 22.172.in-addr.arpa 23.172.in-addr.arpa 24.172.in-addr.arpa 25.172.in-addr.arpa 26.172.in-addr.arpa 27.172.in-addr.arpa 28.172.in-addr.arpa 29.172.in-addr.arpa 30.172.in-addr.arpa 31.172.in-      addr.arpa 168.192.in-addr.arpa d.f.ip6.arpa corp home internal intranet lan local private test
    Aug 18 16:41:04 minikube systemd-resolved[2057]: Using system hostname 'minikube'.
    Aug 18 16:41:04 minikube systemd[1]: Started Network Name Resolution.

**Anything else do we need to know**:

    ~ $ kubectl cluster-info                                                                                                               
    Kubernetes master is running at https://192.168.42.247:8443                                                                            
                                                                                                                                       
    To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
    ~ $ kubectl cluster-info dump                                                                                                          
    The connection to the server 192.168.42.247:8443 was refused - did you specify the right host or port?

Let me know if you need more info!

Thanks,

Scott

**Comment by BrendanBall** [comment]
I've now encountered the same problem. I've been running minikube fine for about 2 weeks now. This morning after I ran `minikube stop` and `minikube start`, localkube didn't start up this time and there are no containers running in the VM. There is also no available network inside the VM. As far as minikube is concerned, it started up all the cluster components correctly, ie. when running `minikube start` it outputs as if everything was successful.

Deleting and recreating minikube at this point doesn't fix the problem, I'm not sure how to fix it at this point. 

I'm basically running the same setup as @firecat53 with arch so it might be specific to the arch setup.
